src/3dDiff.c

```diff
@@ -0,0 +1,407 @@
+#include "mrilib.h"
+#include "3ddata.h"
+
+#define MRI_SIMPLE_DIFF_ERROR -1
+#define MRI_SIMPLE_DIFF_DEFAULT_TOL 1e-8f
+
+#define diff3d_crash(...) \
+    do { \
+        if ( quiet ) { \
+            printf("-1\n"); \
+            RETURN ( -1 ); \
+        } \
+        if ( brutalist ) { \
+            printf("-1 -1 -1 -1\n"); \
+            RETURN ( -1 ); \
+        } \
+        if ( report || long_report ) { \
+            ERROR_exit(__VA_ARGS__); \
+        } \
+    } \
+    while ( 0 )
+
+void repchar(char c, int reps) {
+    for (int i = 0; i < reps; ++i) putchar(c);
+}
+
+int jbt_get_terminal_width() {
+    const int DEFAULT_WIDTH = 79;
+    int width = 0;
+    char * term_width = getenv("COLUMNS");
+    if ( !term_width ) width = DEFAULT_WIDTH;
+    else {
+        width = atoi(term_width);
+        if ( width < 0 ) width = DEFAULT_WIDTH;
+    }
+    return width;
+}
+void dashline() {
+    repchar('-', jbt_get_terminal_width());
+    putchar('\n');
+}
+
+int help_3dDiff()
+{
+char * author = "JB Teves";
+
+printf(
+"\n"
+"This is a program to examine element-wise differences between two images.\n"
+"\n"
+"written by: %s\n"
+"\n"
+"Usage: 3dDiff  [display opt] [-tol TOLERANCE] [-mask MASK] <DSET_1> <DSET_2>\n"
+"\n"
+"where: \n"
+"\n"
+"  -tol TOLERANCE   :(opt) the floating-point tolerance/epsilon\n"
+"\n"
+"  -mask MASK:      :(opt) the mask to use when comparing\n"
+"\n"
+"  -a DSET_1        :(req) input dataset a\n"
+"\n"
+"  -b DSET_2        :(req) input dataset b\n"
+"\n"
+"... and there are the following (mutually exclusive) display options:\n"
+"\n"
+"  -q               :(opt) quiet mode, indicate 0 for no differences and\n"
+"                          1 for differences. (aka \"Rick Mode\")\n"
+"  -tabular         :(opt) display only a table of differences, plus\n"
+"                          a summary line (the same one as -brutalist)\n"
+"  -brutalist       :(opt) display one-liner. The first number indicates\n"
+"                          whether there is a difference, the second number \n"
+"                          indicates how many elements (3D) or volumes (4D)\n"
+"                          were different, and the last number indicates the\n"
+"                          total number of elements/volumes compared.\n"
```

Comment:
these should each describe the output when things don't match---particularly for brutalist and quiet

For tabular, I would provide an explicit example of what you call the "table of differences" part looks like, because it will be hard for the user to parse that couple numbers, esp. when only 1 volume is input, probably.